### PR TITLE
tikv-ctl: try best to keep more logs when unsafe-recover

### LINKED ISSUE #3092: Unsafe recover should only drop some nodes instead of all down nodes
For example, A region contains six nodes a, b, c, d, e and f, a is leader. If a, b and c are down forever, we need to just delete two nodes instead of all three nodes. It's possible that just a, b, c and d share the committed log. If a, b, c are deleted, d's committed log can be overridden if e or f becomes leader. Note that d may even not know that a has committed the log, so the cluster can run smoothly without knowing that committed logs are lost, which can be prevented in this case.

### PR DESCRIPTION
**Pull request by hicqu**
Fix #3092.

### CODE REVIEW

File: src/server/debug.rs [hunk 1]
```diff
@@ -400,6 +400,19 @@ impl Debugger {
         let handle = box_try!(get_cf_handle(self.engines.kv_engine.as_ref(), CF_RAFT));
         let store_ids = HashSet::<u64>::from_iter(store_ids);
 
+        let count_voters = |peers: &[Peer]| peers.iter().filter(|p| !p.get_is_learner()).count();
+
+        // Try best to keep committed logs not be overwritten after
```
Comment (BusyJay) [inline src/server/debug.rs]: You can't know precisely whether a log has been committed. So you should keep as many logs as possible.

**Comment by hicqu** [comment]
@BusyJay I'm confused with `max(old_voters_len - rest_voters_len * 2 + 1, 0)`. For example, if `old_voters_len` is 6 and `rest_voters_len` = 3, we can remove `2` voters but we got `1` from it.

**Comment by BusyJay** [comment]
It's obvious that preserve as many logs as possible implies that committed log won't be overwritten if possible. Assuming the final region has `n` voters, then it's quorum should be `n / 2 + 1`. So the best case is that all remaining voters form the very quorum, so `rest_voters_len = n / 2 + 1`. `n` should be in `{2 * rest_voters_len - 2, 2 * rest_voters_len - 1}`, delete count should be `old_voters_len - 2 * rest_voters_len + 2` or `old_voters_len - 2 * rest_voters_len + 1`. Choosing latter to make less deletions so that cases like 5/3 can be recovered automatically without touching the metadata.

**Comment by BusyJay** [comment]
Any updates?

**Comment by hicqu** [comment]
It's dangerous to decide which stores to remove automaticly because if leader and followers have different peer lists, tikv-ctl could remove not same peers on them.

**Pull request closed without merging**